CNKI.js

```diff
@@ -215,29 +242,88 @@ function scrape(ids, doc, url, itemInfo) {
 				if (!info) {
 					Z.debug('No item info for "' + newItem.title + '"');
 				} else {
-					/*if (!info.pdfURL) {
-						Z.debug('No PDF URL passed from multiples page');
-					} else {
-						newItem.attachments.push({
-							title: 'Full Text PDF',
-							mimeType: 'application/pdf',
-							url: info.pdfURL
-						})
-					}*/
-					
 					newItem.url = info.url;
 				}
 			} else {
 				newItem.url = url;
 			}
-			
-			i++;
+
+			// CN 中国刊物编号，非refworks中的callNumber
+			// CN in CNKI refworks format explains Chinese version of ISSN
+			if (newItem.callNumber){
+			//	newItem.extra = 'CN ' + newItem.callNumber;
+				newItem.callNumber = "";
+			}
+			// don't download PDF/CAJ on searchResult(multiple)
+			if (resultNum == 'single'){
+				newItem.attachments = getAttachments(doc, newItem);
+			} else if (resultNum == 'multiple') {
+				newItem.attachments = [{
+					url: newItem.url,
+					title: newItem.title,
+					mimeType: "text/html",
+					snapshot: true
+				}];
+			}
 			newItem.complete();
 		});
 		
 		translator.translate();
-	})
+	});
+}
+
+// get pdf download link
+function getPDF(doc) {
+	var pdf = ZU.xpath(doc, "//a[@name='pdfDown']");
+	return pdf.length ? pdf[0].href : false;
+}
+
+// caj download link, default is the whole article for thesis.
+function getCAJ(doc, itemType) {
+	// //div[@id='DownLoadParts']
+	if (itemType == 'thesis') {
+		var caj = ZU.xpath(doc, "//div[@id='DownLoadParts']/a");
+	} else {
+		caj = ZU.xpath(doc, "//a[@name='cajDown']");
+	}
+	return caj.length ? caj[0].href : false;
+}
+
+// add pdf or caj to attachments, default is pdf
+function getAttachments(doc, item) {
+	var attachments = [{
+		url: item.url,
+		title: item.title,
+		mimeType: "text/html",
+		snapshot: true
+	}];
```

Comment:
Is it worth saving a snapshot? Generally, a snapshot is useful if 1) the page contains full-text content, particularly if it might later become unavailable (e.g., a newspaper/magazine article that might go behind a paywall) or 2) the page contains text that might be useful for searching. (Here, it looks like the page content is also dynamically loaded, and the way snapshots are currently saved doesn't make saving such pages very useful.)